### PR TITLE
Setup and comply with latest RuboCop

### PR DESCRIPTION
**Pull request by jonallured**
Just a quick lil PR to upgrade RuboCop to latest in order to get my editor to stop complaining, haha. All I really did was remove the pin and then run this:

```
bundle exec rubocop -A --auto-gen-config
```

### CODE REVIEW

**Comment by jonallured** [comment]
@starsirius hrm, looks like maybe forked PRs don't get built?

**Comment by starsirius** [comment]
Just enabled it in CircleCI dashboard. Wanna try again? 🤞

**Comment by jonallured** [comment]
Ok that worked, thanks! CI is failing because it's using the wrong version of RuboCop. This is confusing me because I removed the pin from the Gemfile and the Gemfile.lock is ignored so I would expect for CI to install the latest version of RuboCop but check this one:

https://app.circleci.com/pipelines/github/artsy/money_helper/16/workflows/2508d70c-3a90-49fa-b558-275a58f078fe/jobs/261

RuboCop is complaining but if you look through the Install Dependencies step you'll find this:

```
...
Installing rubocop 0.59.2
...
```

That's weird right? Why didn't it use a newer version? 🤔

**Comment by jonallured** [comment]
Ugh I see it now - each of the individual gemfiles ALSO have the rubocop pin. Will tinker and try again.

File: .rubocop_todo.yml
Comment (starsirius) [inline .rubocop_todo.yml:13]: _I think_ the various *gemfiles might not be violating the 👮‍♂️ s. Can we try removing all todos and see if it breaks things?

File: Gemfile
Comment (starsirius) [inline Gemfile:7]: I wonder if not specifying a version will lead to drift between CI and local development in the future, given the lockfiles are not checked in. I.e. when the next version of rubocop is released,

- CI is in a clean state and `bundle install` will install the latest version.
- Locally devs might already have an older version installed and `bundle install` would not update rubocop to the latest.

**Review by starsirius — APPROVED** [review]
Have some non-blocking comments and LGTM! Feel free to merge when ready! 💯

File: .rubocop_todo.yml
Comment (jonallured) [reply .rubocop_todo.yml]: Huh you're right - I did this:

```
rm .rubocop_todo.yml
bundle exec rubocop -A --auto-gen-config
```

And these parts of the file were removed - not quite sure how that happened. 🤔

File: Gemfile
Comment (jonallured) [reply Gemfile:7]: Yeah this is a good point - how do you think we should approach this? Should I go back to pinning to the most recent version?

Comment (starsirius) [reply Gemfile:7]: Yeah I'm not sure about the best practice, and it seems like some popular gems are at least specifying some versions (e.g. [thor](https://github.com/rails/thor/blob/efe79b599eb401d36265ed4a8b48fb60f42a06a4/Gemfile#L15), [aws-sdk-ruby](https://github.com/aws/aws-sdk-ruby/blob/0f40509d517af9c923b364547ac1bcbaa35862fa/Gemfile#L54)). Maybe we can go back and pin the latest version?

**Comment by jonallured** [comment]
Ok I think I got this all fixed up. I altered the Gemfile pin here so that the appraisal gemfiles all come out with a stable version on CI. Going to merge and then open a PR to update the supported ActiveSupport versions.

**Pull request merged**
